test/test_pumactl.rb

```diff
@@ -29,6 +29,23 @@ def test_config_file
     assert_equal "t3-pid", control_cli.instance_variable_get("@pidfile")
   end
 
+  def test_control_no_token
+    opts = [
+      "--config-file", "test/config/control_no_token.rb",
+      "start"
+    ]
+
+    control_cli = Puma::ControlCLI.new opts, @ready, @ready
+    t = Thread.new do
```

Comment:
The assert passes with the thread being created.

So, should it be removed, or an assert/refute added for it?

Reply:
Sorry I don't understand what you mean. Would you mind elaborating?

Also I think there's a better way to write this test which shouldn't require a thread: I think checking that the value is a String rather than checking that the server boots successfully should be enough.

Reply:
not sure what @MSP-Greg means either but this test is also nondeterministic - control_cli.run may not finish before we run the assertion

Reply:
I think you could just fix that with a .join though

Reply:
I think the assert statement could be moved above the `Thread.new` statement, or, the instance variable is set with `Puma::ControlCLI.new`, not `#run`.  Sorry, gone for most of the morning...

Reply:
I think it's been a mistake trying to test two things in a single test: It was explicitly testing that the value was being set to 'none' and also implicitly testing that the server could boot with `no_token: true`.

I've removed the part about booting the server.